File: Python/String/Camel_Case.py
#Name :  Atul Kumar
#Github username : atul1510
#Repositary name : Algorithms
#Problem Description
#Program to convert  text sentence in CamelCase format .
#Examples:
 #1] INPUT: "this is camel case"
    #OUTPUT "thisIsCamelCase"
 #2] INPUT: "Hi Atul here"
  # OUTPUT: "hiAtulHere"
   #Example Explanation

# Main Function
def main():
    N = (int(input("No of TestCases : ")))
    for _ in range(N):
        S = input("Enter a sentence : ")
        print("Original: {}".format(S))
        S = S.title() #Convert the first only the first letter of a word to uppercase and rest all to lowercase
        S = S.split() #Generates a list of words
        S[0] = S[0].lower()
        S = "".join(S) #Joins all the words
        print("Camel Casing : {}".format(S))
        print()

if __name__=="__main__":
    main()

File: C++/String/Camel_Case.cpp
/*Name :  Atul Kumar
Github username : atul1510
Repositary name : Algorithms
*/
/*Problem Description
Program to convert  text sentence in CamelCase format .
Examples:
 1] INPUT: "this is camel case"
    OUTPUT "thisIsCamelCase"
 2] INPUT: "Hi Atul here"
   OUTPUT: "hiAtulHere"
Example Explanation */

#include <iostream>
#include <string>
#include <cctype>
using namespace std;

int main() {
    int Testcase;
    cout << "Enter no of Testcases: ";
    cin >> Testcase;
    cout << endl;
    cin.ignore(); // Clear the input buffer
    while (Testcase--) {
        string str;
        cout << "Enter a sentence: ";
        getline(cin, str);
        bool flag = false;
        string camelCaseString = "";
        for (int i = 0; i < str.length(); i++) {
            if (i == 0) camelCaseString += tolower(str[i]);   //Converts to lowercase
            else if (str[i] == ' ') flag = true; // Check for spaces in the sentence
            else if (flag && str[i] != ' ') {
                camelCaseString += toupper(str[i]);   //Converts to uppercase
                flag = false;
            }
            else camelCaseString += tolower(str[i]);   //Converts to lowercase
        }
        cout << "Original: " << str << endl;
        cout << "Camel Case: " << camelCaseString << endl;
    }
    return 0;
}